<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>DlgJoint</class>
 <widget class="QWidget" name="DlgJoint">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>499</width>
    <height>671</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>Rigid Body Joint</string>
  </property>
  <layout class="QVBoxLayout" name="verticalLayout_2">
   <item>
    <widget class="QGroupBox" name="props">
     <property name="title">
      <string>Joint properties</string>
     </property>
     <layout class="QGridLayout" name="gridLayout_2">
      <item row="2" column="1">
       <widget class="QComboBox" name="cmbType"/>
      </item>
      <item row="3" column="0">
       <widget class="QLabel" name="label_3">
        <property name="text">
         <string>Node A</string>
        </property>
       </widget>
      </item>
      <item row="1" column="0">
       <widget class="QLabel" name="label">
        <property name="text">
         <string>Name</string>
        </property>
       </widget>
      </item>
      <item row="1" column="1">
       <widget class="QLineEdit" name="jointName">
        <property name="readOnly">
         <bool>false</bool>
        </property>
       </widget>
      </item>
      <item row="6" column="0">
       <widget class="QLabel" name="label_4">
        <property name="text">
         <string>Node B</string>
        </property>
       </widget>
      </item>
      <item row="0" column="0">
       <widget class="QLabel" name="label_8">
        <property name="text">
         <string>ID</string>
        </property>
       </widget>
      </item>
      <item row="2" column="0">
       <widget class="QLabel" name="label_2">
        <property name="text">
         <string>Type</string>
        </property>
       </widget>
      </item>
      <item row="6" column="1">
       <widget class="QComboBox" name="cmbDstNode"/>
      </item>
      <item row="0" column="1">
       <widget class="QLineEdit" name="jointID">
        <property name="readOnly">
         <bool>true</bool>
        </property>
       </widget>
      </item>
      <item row="3" column="1">
       <widget class="QComboBox" name="cmbSrcNode"/>
      </item>
     </layout>
    </widget>
   </item>
   <item>
    <widget class="QGroupBox" name="groupBox">
     <property name="title">
      <string>Joint parameters</string>
     </property>
     <layout class="QVBoxLayout" name="verticalLayout">
      <item>
       <widget class="QWidget" name="widget" native="true">
        <layout class="QGridLayout" name="gridLayout">
         <item row="12" column="0">
          <widget class="QLabel" name="lblMaxDistance">
           <property name="text">
            <string>Max distance</string>
           </property>
          </widget>
         </item>
         <item row="7" column="1">
          <widget class="QDoubleSpinBox" name="motorTorque">
           <property name="suffix">
            <string> N⋅m</string>
           </property>
          </widget>
         </item>
         <item row="6" column="1">
          <widget class="QDoubleSpinBox" name="motorSpeed">
           <property name="suffix">
            <string> rad/s</string>
           </property>
           <property name="minimum">
            <double>0.000000000000000</double>
           </property>
           <property name="maximum">
            <double>100.000000000000000</double>
           </property>
          </widget>
         </item>
         <item row="2" column="1">
          <widget class="QDoubleSpinBox" name="upperAngle">
           <property name="suffix">
            <string>  °</string>
           </property>
           <property name="minimum">
            <double>0.000000000000000</double>
           </property>
           <property name="maximum">
            <double>360.000000000000000</double>
           </property>
           <property name="stepType">
            <enum>QAbstractSpinBox::AdaptiveDecimalStepType</enum>
           </property>
          </widget>
         </item>
         <item row="7" column="0">
          <widget class="QLabel" name="lblMotorTorque">
           <property name="text">
            <string>Motor torque</string>
           </property>
          </widget>
         </item>
         <item row="0" column="1">
          <layout class="QHBoxLayout" name="nodeAAnchor">
           <item>
            <widget class="QDoubleSpinBox" name="srcX">
             <property name="enabled">
              <bool>true</bool>
             </property>
             <property name="prefix">
              <string>x = </string>
             </property>
             <property name="minimum">
              <double>-1000000.000000000000000</double>
             </property>
             <property name="maximum">
              <double>1000000.000000000000000</double>
             </property>
             <property name="stepType">
              <enum>QAbstractSpinBox::AdaptiveDecimalStepType</enum>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QDoubleSpinBox" name="srcY">
             <property name="enabled">
              <bool>true</bool>
             </property>
             <property name="buttonSymbols">
              <enum>QAbstractSpinBox::UpDownArrows</enum>
             </property>
             <property name="prefix">
              <string>y = </string>
             </property>
             <property name="minimum">
              <double>-1000000.000000000000000</double>
             </property>
             <property name="maximum">
              <double>1000000.000000000000000</double>
             </property>
             <property name="stepType">
              <enum>QAbstractSpinBox::AdaptiveDecimalStepType</enum>
             </property>
            </widget>
           </item>
          </layout>
         </item>
         <item row="14" column="0">
          <widget class="QLabel" name="lblStiffness">
           <property name="text">
            <string>Stiffness</string>
           </property>
          </widget>
         </item>
         <item row="0" column="2">
          <widget class="QToolButton" name="btnResetSrcAnchor">
           <property name="toolTip">
            <string>Reset idle track to nothing</string>
           </property>
           <property name="text">
            <string>...</string>
           </property>
           <property name="icon">
            <iconset resource="../resource/resource.qrc">
             <normaloff>:/16x16_ico_png/reset.png</normaloff>:/16x16_ico_png/reset.png</iconset>
           </property>
           <property name="autoRaise">
            <bool>false</bool>
           </property>
           <property name="arrowType">
            <enum>Qt::NoArrow</enum>
           </property>
          </widget>
         </item>
         <item row="0" column="0">
          <widget class="QLabel" name="lblANodeAnchor">
           <property name="text">
            <string>Node A anchor at</string>
           </property>
          </widget>
         </item>
         <item row="1" column="0">
          <widget class="QLabel" name="lblBNodeAnchor">
           <property name="text">
            <string>Node B anchor at</string>
           </property>
          </widget>
         </item>
         <item row="10" column="0">
          <widget class="QLabel" name="lblMinDistance">
           <property name="text">
            <string>Min distance</string>
           </property>
          </widget>
         </item>
         <item row="3" column="0">
          <widget class="QLabel" name="lblLowerAngleLimit">
           <property name="text">
            <string>Max rotation CW</string>
           </property>
           <property name="wordWrap">
            <bool>false</bool>
           </property>
          </widget>
         </item>
         <item row="14" column="1">
          <widget class="QDoubleSpinBox" name="stiffness">
           <property name="suffix">
            <string> N/m</string>
           </property>
          </widget>
         </item>
         <item row="1" column="2">
          <widget class="QToolButton" name="btnResetDstAnchor">
           <property name="toolTip">
            <string>Reset idle track to nothing</string>
           </property>
           <property name="text">
            <string>...</string>
           </property>
           <property name="icon">
            <iconset resource="../resource/resource.qrc">
             <normaloff>:/16x16_ico_png/reset.png</normaloff>:/16x16_ico_png/reset.png</iconset>
           </property>
           <property name="autoRaise">
            <bool>false</bool>
           </property>
           <property name="arrowType">
            <enum>Qt::NoArrow</enum>
           </property>
          </widget>
         </item>
         <item row="15" column="1">
          <widget class="QDoubleSpinBox" name="damping">
           <property name="prefix">
            <string/>
           </property>
           <property name="suffix">
            <string> N⋅s/m</string>
           </property>
          </widget>
         </item>
         <item row="9" column="1">
          <widget class="QCheckBox" name="chkEnableLimit">
           <property name="text">
            <string>Enable rotation limits</string>
           </property>
          </widget>
         </item>
         <item row="1" column="1">
          <layout class="QHBoxLayout" name="nodeBAnchor">
           <item>
            <widget class="QDoubleSpinBox" name="dstX">
             <property name="enabled">
              <bool>true</bool>
             </property>
             <property name="prefix">
              <string>x = </string>
             </property>
             <property name="minimum">
              <double>-1000000.000000000000000</double>
             </property>
             <property name="maximum">
              <double>1000000.000000000000000</double>
             </property>
             <property name="stepType">
              <enum>QAbstractSpinBox::AdaptiveDecimalStepType</enum>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QDoubleSpinBox" name="dstY">
             <property name="enabled">
              <bool>true</bool>
             </property>
             <property name="buttonSymbols">
              <enum>QAbstractSpinBox::UpDownArrows</enum>
             </property>
             <property name="prefix">
              <string>y = </string>
             </property>
             <property name="minimum">
              <double>-1000000.000000000000000</double>
             </property>
             <property name="maximum">
              <double>1000000.000000000000000</double>
             </property>
             <property name="stepType">
              <enum>QAbstractSpinBox::AdaptiveDecimalStepType</enum>
             </property>
            </widget>
           </item>
          </layout>
         </item>
         <item row="2" column="0">
          <widget class="QLabel" name="lblUpperangleLimit">
           <property name="text">
            <string>Max rotation CCW</string>
           </property>
           <property name="scaledContents">
            <bool>false</bool>
           </property>
           <property name="wordWrap">
            <bool>false</bool>
           </property>
          </widget>
         </item>
         <item row="10" column="2">
          <widget class="QToolButton" name="btnResetMinDistance">
           <property name="toolTip">
            <string>Reset idle track to nothing</string>
           </property>
           <property name="text">
            <string>...</string>
           </property>
           <property name="icon">
            <iconset resource="../resource/resource.qrc">
             <normaloff>:/16x16_ico_png/reset.png</normaloff>:/16x16_ico_png/reset.png</iconset>
           </property>
           <property name="autoRaise">
            <bool>false</bool>
           </property>
           <property name="arrowType">
            <enum>Qt::NoArrow</enum>
           </property>
          </widget>
         </item>
         <item row="8" column="1">
          <widget class="QCheckBox" name="chkEnableMotor">
           <property name="text">
            <string>Enable motor</string>
           </property>
          </widget>
         </item>
         <item row="15" column="0">
          <widget class="QLabel" name="lblDamping">
           <property name="text">
            <string>Damping</string>
           </property>
          </widget>
         </item>
         <item row="12" column="1">
          <widget class="QDoubleSpinBox" name="maxDist">
           <property name="enabled">
            <bool>true</bool>
           </property>
           <property name="specialValueText">
            <string>Initial distance between nodes</string>
           </property>
           <property name="decimals">
            <number>1</number>
           </property>
           <property name="minimum">
            <double>-0.100000000000000</double>
           </property>
           <property name="maximum">
            <double>1000000.000000000000000</double>
           </property>
           <property name="stepType">
            <enum>QAbstractSpinBox::AdaptiveDecimalStepType</enum>
           </property>
           <property name="value">
            <double>-0.100000000000000</double>
           </property>
          </widget>
         </item>
         <item row="10" column="1">
          <widget class="QDoubleSpinBox" name="minDist">
           <property name="enabled">
            <bool>true</bool>
           </property>
           <property name="specialValueText">
            <string>Initial distance between nodes</string>
           </property>
           <property name="decimals">
            <number>1</number>
           </property>
           <property name="minimum">
            <double>-0.100000000000000</double>
           </property>
           <property name="maximum">
            <double>1000000.000000000000000</double>
           </property>
           <property name="stepType">
            <enum>QAbstractSpinBox::AdaptiveDecimalStepType</enum>
           </property>
           <property name="value">
            <double>-0.100000000000000</double>
           </property>
          </widget>
         </item>
         <item row="6" column="0">
          <widget class="QLabel" name="lblMotorSpeed">
           <property name="text">
            <string>Motor speed</string>
           </property>
          </widget>
         </item>
         <item row="12" column="2">
          <widget class="QToolButton" name="btnResetMaxDistance">
           <property name="toolTip">
            <string>Reset idle track to nothing</string>
           </property>
           <property name="text">
            <string>...</string>
           </property>
           <property name="icon">
            <iconset resource="../resource/resource.qrc">
             <normaloff>:/16x16_ico_png/reset.png</normaloff>:/16x16_ico_png/reset.png</iconset>
           </property>
           <property name="autoRaise">
            <bool>false</bool>
           </property>
           <property name="arrowType">
            <enum>Qt::NoArrow</enum>
           </property>
          </widget>
         </item>
         <item row="3" column="1">
          <widget class="QDoubleSpinBox" name="lowerAngle">
           <property name="suffix">
            <string>  °</string>
           </property>
           <property name="minimum">
            <double>0.000000000000000</double>
           </property>
           <property name="maximum">
            <double>360.000000000000000</double>
           </property>
           <property name="stepType">
            <enum>QAbstractSpinBox::AdaptiveDecimalStepType</enum>
           </property>
          </widget>
         </item>
         <item row="16" column="1">
          <spacer name="verticalSpacer">
           <property name="orientation">
            <enum>Qt::Vertical</enum>
           </property>
           <property name="sizeHint" stdset="0">
            <size>
             <width>20</width>
             <height>40</height>
            </size>
           </property>
          </spacer>
         </item>
         <item row="4" column="1">
          <widget class="QComboBox" name="cmbMotorRotation">
           <item>
            <property name="text">
             <string>Counterclockwise</string>
            </property>
           </item>
           <item>
            <property name="text">
             <string>Clockwise</string>
            </property>
           </item>
          </widget>
         </item>
         <item row="4" column="0">
          <widget class="QLabel" name="lblMotorRotation">
           <property name="text">
            <string>Motor rotation</string>
           </property>
          </widget>
         </item>
        </layout>
       </widget>
      </item>
     </layout>
    </widget>
   </item>
   <item>
    <layout class="QHBoxLayout" name="horizontalLayout">
     <item>
      <spacer name="horizontalSpacer">
       <property name="orientation">
        <enum>Qt::Horizontal</enum>
       </property>
       <property name="sizeHint" stdset="0">
        <size>
         <width>40</width>
         <height>20</height>
        </size>
       </property>
      </spacer>
     </item>
     <item>
      <widget class="QPushButton" name="btnAccept">
       <property name="text">
        <string>OK</string>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QPushButton" name="btnApply">
       <property name="text">
        <string>Apply</string>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QPushButton" name="btnCancel">
       <property name="text">
        <string>Cancel</string>
       </property>
      </widget>
     </item>
    </layout>
   </item>
  </layout>
 </widget>
 <tabstops>
  <tabstop>jointID</tabstop>
  <tabstop>jointName</tabstop>
  <tabstop>cmbType</tabstop>
  <tabstop>cmbSrcNode</tabstop>
  <tabstop>cmbDstNode</tabstop>
  <tabstop>srcX</tabstop>
  <tabstop>srcY</tabstop>
  <tabstop>btnResetSrcAnchor</tabstop>
  <tabstop>dstX</tabstop>
  <tabstop>dstY</tabstop>
  <tabstop>btnResetDstAnchor</tabstop>
  <tabstop>upperAngle</tabstop>
  <tabstop>lowerAngle</tabstop>
  <tabstop>cmbMotorRotation</tabstop>
  <tabstop>motorSpeed</tabstop>
  <tabstop>motorTorque</tabstop>
  <tabstop>chkEnableMotor</tabstop>
  <tabstop>chkEnableLimit</tabstop>
  <tabstop>minDist</tabstop>
  <tabstop>btnResetMinDistance</tabstop>
  <tabstop>maxDist</tabstop>
  <tabstop>btnResetMaxDistance</tabstop>
  <tabstop>stiffness</tabstop>
  <tabstop>damping</tabstop>
  <tabstop>btnAccept</tabstop>
  <tabstop>btnApply</tabstop>
  <tabstop>btnCancel</tabstop>
 </tabstops>
 <resources>
  <include location="../resource/resource.qrc"/>
 </resources>
 <connections/>
</ui>
